bee-storage/bee-storage/src/lib.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A general purpose storage backend crate with key value abstraction API.
+//!
+//! # Features
+//!
+//! ## Access implementation:
+//! - Traits contracts which define the general purpose database operations such as (insert, fetch, ...);
+//! ## Backend implementation:
+//! - Trait contract to start and shutdown backends;
+//! - Configuration and associated builder to configure different backends;
+//!
+//! This crate tries to simplify the implementation of various storage backends and provides unified access API for the
+//! application/user space.
+
+#![deny(missing_docs)]
```

Comment:
I'ld also include `#![deny(warnings)]`, bc I think for a production library this should be the default. We can still explicitly allow things that would throw a warning, but it would force us to think about why a warning is okay in a particular situation.

Reply:
Sure why not. The CI is actually already doing it repository wise so this is not strictly necessary but it can't hurt to add it.